docs/developer/adr/0006-add-readonly-flags.rst

```diff
@@ -0,0 +1,66 @@
+ADR 0006: Add read-only flags
+=============================
+
+- Status: accepted
+- Deciders: Jan-Lukas, Neil, Owen, Simon
+- Date: 2021-05-05
+
+Context
+-------
+
+There are a number of contexts where values of variables are conceptually "broadcast" (not necessarily using an actual ``broadcast`` operation) and are thus shared (not to be confused with sharing from the shallow-copy mechanism):
+
+- Explicit ``broadcast`` operations.
+- Masks of dimension ``N-M`` if data is of dimension ``N``.
+  Each mask value is conceptually shared by all data values along the ``M`` missing dimensions in the mask.
+- Coords along dimension ``dim0`` of slices of a data array along dimension ``dim1``.
+  The coord values are conceptually shared by all slices.
+- Items independent of ``dim`` in a dataset which is then sliced along ``dim``.
+  These items are conceptually shared by all slices.
+- Coords of items in a dataset.
+  The coords are conceptually shared by all items.
+
+In all of the above cases in-place a subsequent modifications would silently affect other unrelated (sub)objects such as other slices or items of the same "parent" object.
```

Comment:
Typo, change to, e.g.,
`In all of the above cases, subsequent modifications`
(not sure about the comma)